### PR TITLE
Dependency Update and Fixes

### PR DESCRIPTION
**Pull request by wSedlacek**
## Description

This updates the dependencies and fixes conflicts introduced with their updates.

fixes: #30

### CODE REVIEW

File: src/lib/prisma-session-store.spec.ts [hunk 1]
```diff
@@ -6,6 +6,15 @@ import { createPrismaMock } from '../mocks';
...
 
+declare module 'express-session' {
+  // tslint:disable-next-line: naming-convention
+  interface SessionData {
+    sample?: boolean;
+    unrealizable?: string;
+    data?: string;
+  }
+}
+
```
Comment (wSedlacek) [inline src/lib/prisma-session-store.spec.ts:17]: With the latest type definitions of `@types/express-session` this is how `SessionData` has it's types defined.
See: [@types/express-session/express-session-tests.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/da76ba1925ec673e67a26a3ddf5e43965d7ab185/types/express-session/express-session-tests.ts#L63-L68)

File: src/lib/prisma-session-store.ts [hunk 2]
```diff
@@ -425,46 +463,4 @@ export class PrismaSessionStore extends Store {
...
-    ).catch(() => {
-      this.disable();
-      this.stopInterval();
-      this.logger.error(dedent`Could not connect to Sessions model in Prisma.
-      Please make sure that prisma is setup correctly and that your migrations are current.
-      For more information check out https://github.com/kleydon/prisma-session-store`);
-    });
-
-    return !this.invalidConnection;
-  }
```
Comment (wSedlacek) [inline src/lib/prisma-session-store.ts:469]: These were moved due to updates to the linting `member-order`

File: yarn.lock [hunk 3]
```diff
@@ -2,10 +2,10 @@
 # yarn lockfile v1
```
Comment (wSedlacek) [inline yarn.lock:2]: This needed to be recreated because of a breaking change with `@types/express` that required it to be reinstalled.

**Comment by wSedlacek** [comment]
@kleydon When you get a chance, take look at this. It fixes some typing changes and updates the packages.
Some of these changes are breaking due to the fact that `@types/express` and `@types/express-session` are not dependencies.

We should probably discuss locking our package versions at some point.
We should also figure out what is the best way to assure compatible `@types/express-session` and `@types/express` is used. Typically I would use `peerDependencies` but I am not sure that makes sense for types.

**Comment by kleydon** [comment]
@wSedlacek - thanks for having a look at this! The changes seem solid, as far as I can tell. 

>We should probably discuss locking our package versions at some point.

Do you mean in the sense of providing a lock file for each package version? Or in the sense of "locking" a version of the library to a particular version (or set of versions) of express / express-session? (Maybe it amounts to the same thing?)

>We should also figure out what is the best way to assure compatible @types/express-session and @types/express is used. Typically I would use peerDependencies but I am not sure that makes sense for types.

Hmmm... I'm not sure either. What do you see as the down-sides for using peer dependencies? Do any alternatives that pop to mind? Would use of peerDependencies only dictate the version of express / express-session used _by this lib_, or would it effectively dictate the version of express / express-session that a developer could use in their own application code when using this lib?

It seems conceivable that we could try to support older _and_ newer typings - but I'm definitely not thrilled about that prospect...

A related thought: Is there any known, straightforward way for libraries like this one to automatically run tests whenever dependencies (i.e. express, express-session, or their corresponding @types/ ) are updated?

**Review by kleydon — APPROVED** [review]
@wSedlacek - thanks for having a look at this! The changes seem solid, as far as I can tell.

>We should probably discuss locking our package versions at some point.

Do you mean in the sense of providing a lock file for each package version? Or in the sense of "locking" a version of the library to a particular version (or set of versions) of express / express-session? (Maybe it amounts to the same thing?)

>We should also figure out what is the best way to assure compatible @types/express-session and @types/express is used. Typically I would use peerDependencies but I am not sure that makes sense for types.

Hmmm... I'm not sure either. What do you see as the down-sides for using peer dependencies? Do any alternatives that pop to mind? Would use of peerDependencies only dictate the version of express / express-session used by this lib, or would it effectively dictate the version of express / express-session that a developer could use in their own application code when using this lib?

It seems conceivable that we could try to support older and newer typings - but I'm definitely not thrilled about that prospect...

Related thought: Is there any known, straightforward way for libraries like this one to automatically run tests whenever dependencies (i.e. express, express-session, or their corresponding @types/ ) are updated?

**Comment by kleydon** [comment]
@wSedlacek Given that there are breaking changes here: What do you think about addressing the `expires` -> `expiresAt` consistency issue, in conjunction? Do you think it would it make sense to "lump" these dislocations into one set of breaking changes - or do you think it would confuse the issue / complicate things?

**Comment by wSedlacek** [comment]
As far as locking, I mean for Prisma and express-session. They are currently set to * which was good when Prisma was updating so frequently without breaking changes but at this point older versions of Prisma won't work and future versions may or may not work so it might be time to be more strict about the versioning.

Supporting older typings seems fruitless. Our primary two dependencies, Prisma and Express-session, shouldn't have a ton of other dependencies in the way that a framework like angular or react would so it is quite unlikely that someone would be stuck on an older version of one of those dependencies and still want the latest version of our library. And even if we do want to support this I think it should be done though multiple branches that each correspond to the version of the library they are tracking. Ie our version 2.14.x would be tracking Prisma 2.14.x and express-session 1.7.x and then on the next prism release we would update our express-session dependency as well and support two separate branches for X months. (take a look at the angular release model and how they track the type script supported version for my inspiration)

In any case, supporting multiple versions of dependant packages is a lot of work and if we are going to pursue that then we would likely want at least one other contributor. I don't think it's in the card for us right now. 

As far as assuring type dependencies versions, I haven't seen any libraries do this before so we likely will need to do some research on how to do this properly. I don't have anymore insight into this at this time, but maybe we should open an issue and mark it as help wanted to further discuss it with the community. Using peerDependencies uses the same version in downstream application and provides a warning to the developer if they don't match. Where as a dependency would use a locked version for our package but allow other packages in the project to use different versions. It could potentially cause conflicts I believe. 

Running test against updated dependencies could be done via dependbot or something similar that makes PR to update your dependencies. I believe by default it is setup for just security updates but there might be a way to change that to any version update. 

As far as the breaking change, this change purely relates to the dependencies and anyone updating will need to update their typing packages for express and express-session. No changes in their code should be necessary, at least not for our library, but `@types/express` may need to be removed completely and installed again OR the package-lock file may need to be regenerated. Given the nature of this particular breaking change, I don't think breaking changes involving code changes needed by the developer are warranted. I more or less want to wait until we need to have the developer change their prisma.schema for something else, and this particular change doesn't require that.

**Comment by wSedlacek** [comment]
I am going to go ahead and release these changes.

**Comment by kleydon** [comment]
@wSedlacek Thanks for doing the release.

**Pull request merged**